Bottom: c1d1f17ec0fa2d4a9d6d149d6f51232fbe9153b9
Top:    dbd6a1d8fbe11aa93710e78fe83a8c8ccc13f715
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-01-06 17:05:32 +0000

polygon.c: For some reason, the range computation in frac_circle appears to be wrong


---

diff --git a/src/polygon.c b/src/polygon.c
index acd55c5..91a16a8 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -483,8 +483,8 @@ frac_circle (PLINE * c, Coord X, Coord Y, Vector v, int fraction)
   e1 = (v[0] - X) * POLY_CIRC_RADIUS_ADJ;
   e2 = (v[1] - Y) * POLY_CIRC_RADIUS_ADJ;
 
-  /* NB: the caller adds the last vertex, hence the -1 */
-  range = POLY_CIRC_SEGS / fraction - 1;
+  /* XXX */ /* NB: the caller adds the last vertex, hence the -1 */
+  range = POLY_CIRC_SEGS / fraction;
   for (i = 0; i < range; i++)
     {
       /* rotate the vector */
